server/routerlicious/packages/protocol-definitions/src/summary.ts

```diff
@@ -53,6 +57,9 @@ export interface ISummaryTree {
 
     // TODO type I can infer from SummaryObject. File mode I may want to directly specify so have symlink+exec access
     tree: { [path: string]: SummaryObject };
+
+    // Tells the reference state of this tree. If this flag is not present, the tree is considered referenced.
+    referenceFlag?: ReferenceFlag;
```

Comment:
Should this be an integer so that we can add multiple enum values to the same property? That's what a flag really look like IMO.

Reply:
Can it be something like
```
unreferenced?: true;
```
I.e. only allow two states - either it's true, or it's not even present (which means it's false).
I'd not go for bitfields or enums here - there does not seem like a need for it.

Reply:
The reason I made this an enum is because server folks preferred it over boolean. However, as per our discussion, that can be done by the server /driver. We can use a boolean internally in the runtime.